### ISSUE TITLE
cmd,core: RPC server started before p2p node instantiated

### ISSUE THREAD

**Issue by fabioberger** [0xProject/0x-mesh#558]
The RPC server is started before `app.Start(ctx)` completes. `Start()` sets up the OrderWatcher and p2p Node amongst other things. 

Regression introduced in https://github.com/0xProject/0x-mesh/pull/312 where `app.Start(ctx)` went from being a blocking to non-blocking call. 

### Additional changes that aggravated the issue are:
- The introduction of the `mesh_getStats` RPC call that requires fetching state from the p2p node
- Sharing orders immediately over Gossipsub when `mesh_addOrders` is called.
- Adding a blocking call to `blockWatcher.SyncToLatestBlock(innerCtx)` before the BlockWatcher is started within `app.Start`.

### Repro steps

Simply add `time.Sleep(1 * time.Minute)` to `app.Start` and send a `mesh_getStats` or `mesh_addOrders` RPC request.

### Sample error log

```
{"error_string":"runtime error: invalid memory address or nil pointer dereference","level":"info","method_string":"GetStats","msg":"RPC method handler crashed","myPeerID":"16Uiu2HAmDvGt9g9VGjhuowpSadS1nsb6PWimENjV37TaBfCxFD3S","stackTrace_string":"
goroutine 9 [running]:
runtime/debug.Stack(0x4a05140, 0x49f7b60, 0x56b40c0)
    /usr/local/opt/go/libexec/src/runtime/debug/stack.go:24 +0x9d
main.(*rpcHandler).GetStats.func1(0xc0002759b0)
    /Users/rickmorty/Documents/projects/0x/go/src/github.com/0xProject/0x-mesh/cmd/mesh/rpc_handler.go:166 +0xa0
panic(0x49f7b60, 0x56b40c0)
    /usr/local/opt/go/libexec/src/runtime/panic.go:522 +0x1b5
github.com/0xProject/0x-mesh/p2p.(*Node).GetNumPeers(...)
    /Users/rickmorty/Documents/projects/0x/go/src/github.com/0xProject/0x-mesh/p2p/node.go:374
github.com/0xProject/0x-mesh/core.(*App).GetStats(0xc000102b40, 0x4cbede0, 0xc0002759b0, 0x1)
    /Users/rickmorty/Documents/projects/0x/go/src/github.com/0xProject/0x-mesh/core/core.go:805 +0x44d
main.(*rpcHandler).GetStats(0xc0000daaf8, 0x0, 0x0, 0x0)
    /Users/rickmorty/Documents/projects/0x/go/src/github.com/0xProject/0x-mesh/cmd/mesh/rpc_handler.go:171 +0xc0
github.com/0xProject/0x-mesh/rpc.(*rpcService).GetStats(0xc000032a90, 0x0, 0x0, 0x0)
    /Users/rickmorty/Documents/projects/0x/go/src/github.com/0xProject/0x-mesh/rpc/service.go:168 +0x33
reflect.Value.call(0xc0000d92c0, 0xc0000100e8, 0x13, 0x4b2966b, 0x4, 0xc00033e120, 0x1, 0x2, 0xc00008bc98, 0x18, ...)
    /usr/local/opt/go/libexec/src/reflect/value.go:447 +0x461
reflect.Value.Call(0xc0000d92c0, 0xc0000100e8, 0x13, 0xc00033e120, 0x1, 0x2, 0x0, 0x0, 0x4add540)
    /usr/local/opt/go/libexec/src/reflect/value.go:308 +0xa4
github.com/0xProject/0x-mesh/vendor/github.com/ethereum/go-ethereum/rpc.(*callback).call(0xc0000d9320, 0x4e0ace0, 0xc000368180, 0xc0003b27c0, 0xd, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
    /Users/rickmorty/Documents/projects/0x/go/src/github.com/0xProject/0x-mesh/vendor/github.com/ethereum/go-ethereum/rpc/service.go:209 +0x2a3
github.com/0xProject/0x-mesh/vendor/github.com/ethereum/go-ethereum/rpc.(*handler).runMethod(0xc000240090, 0x4e0ace0, 0xc000368180, 0xc00042c070, 0xc0000d9320, 0x0, 0x0, 0x0, 0x0)
    /Users/rickmorty/Documents/projects/0x/go/src/github.com/0xProject/0x-mesh/vendor/github.com/ethereum/go-ethereum/rpc/handler.go:369 +0x8a
github.com/0xProject/0x-mesh/vendor/github.com/ethereum/go-ethereum/rpc.(*handler).handleCall(0xc000240090, 0xc00033e0c0, 0xc00042c070, 0x203000)
    /Users/rickmorty/Documents/projects/0x/go/src/github.com/0xProject/0x-mesh/vendor/github.com/ethereum/go-ethereum/rpc/handler.go:331 +0x231
github.com/0xProject/0x-mesh/vendor/github.com/ethereum/go-ethereum/rpc.(*handler).handleCallMsg(0xc000240090, 0xc00033e0c0, 0xc00042c070, 0x4dfe001)
    /Users/rickmorty/Documents/projects/0x/go/src/github.com/0xProject/0x-mesh/vendor/github.com/ethereum/go-ethereum/rpc/handler.go:298 +0x22d
github.com/0xProject/0x-mesh/vendor/github.com/ethereum/go-ethereum/rpc.(*handler).handleMsg.func1(0xc00033e0c0)
    /Users/rickmorty/Documents/projects/0x/go/src/github.com/0xProject/0x-mesh/vendor/github.com/ethereum/go-ethereum/rpc/handler.go:139 +0x46
github.com/0xProject/0x-mesh/vendor/github.com/ethereum/go-ethereum/rpc.(*handler).startCallProc.func1(0xc000240090, 0xc0000ea7e0)
    /Users/rickmorty/Documents/projects/0x/go/src/github.com/0xProject/0x-mesh/vendor/github.com/ethereum/go-ethereum/rpc/handler.go:226 +0xdf
created by github.com/0xProject/0x-mesh/vendor/github.com/ethereum/go-ethereum/rpc.(*handler).startCallProc
    /Users/rickmorty/Documents/projects/0x/go/src/github.com/0xProject/0x-mesh/vendor/github.com/ethereum/go-ethereum/rpc/handler.go:222 +0x66
","time":"2019-11-22T10:44:00+01:0
"}
```

**Comment by fabioberger** [comment]
From @albrow: "This should be a really easy fix. I would suggest a sight modification to all of our async services (anything with a separate Start method). Any methods that won't work unless/until the service is started should simply block until it is started. We could use a channel called isStarted or something and just close the channel to signal any waiters all at once."

**Comment by albrow** [comment]
Fixed in #560.
